src/chapters/razor-views/template-iteration.rst

```diff
@@ -0,0 +1,247 @@
+      {
+         <li>@collectionItem</li>
+      }
+   </ol>
+
+#. The ``@foreach`` loop is initiated inside of a list element (either ``<ol>`` or ``<ul>``).
+#. ``ViewBag.collectionProperty`` represents any collection that has been assigned as a property on ``ViewBag``.
+#. ``collectionItem`` represents an individual item or element within the collection.
+#. The ``@`` specifies the C# portion of the template.
+
+We can think of this syntax as saying, *"For each item within
+ViewBag's property, 'collectionProperty', repeat this HTML tag, but use the current value of
+'collectionItem'."*
+
+Let's apply this new concept to the HTML coffee list example. Assume that we store each of
+the coffee names as strings in a ``List`` called ``ViewBag.coffeeOptions``.
+
+.. sourcecode:: guess
+   :linenos:
+
+   <ol>
+      @foreach (string coffeeType in ViewBag.coffeeOptions)
+      {
+         <li>@coffeeType</li>
+      }
+   </ol>
+
+Some points to note:
+
+#. ``coffeeOptions`` is accessible to the template because it is a property of the ``ViewBag`` object.
+#. In the first pass through the loop, ``coffeeType`` takes the value of the first
+   coffee name in the ``coffeeOptions`` list.
+#. ``@coffeeType`` displays the value of ``coffeeType`` in the view, so the
+   ``li`` element will show the first coffee name.
+#. Each successive iteration, ``coffeeType`` takes the next value in the list, and
+   Razor adds a new ``<li></li>`` element to display that data.
+
+
+.. _location-matters:
+
+.. admonition:: Warning
+
+   ``@foreach`` creates one HTML tag for each item in a collection. 
+   BE CAREFUL where you place it.
+
+   Consider the following code blocks:
+
+   .. admonition:: Examples
+
+      Option #1:
+
+      .. sourcecode:: guess
+         :linenos:
+
+         <div>
+            <h3>C# Types</h3>
+            <ol>
+               @foreach (string coffeeType in ViewBag.coffeeTypes)
+               {
+                  <li>@coffeeType</li>
+               }
+            </ol>
+         </div>
+
+      Option #2:
+
+      .. sourcecode:: guess
+         :linenos:
+
+         <div>
+            <h3>C# Types</h3>
+            @foreach (string coffeeType in ViewBag.coffeeTypes)
+            {
+               <ol>
+                  <li>@coffeeType</li>
+               </ol>
+            }
+         </div>
+
+      Option #3:
+
+      .. sourcecode:: guess
+         :linenos:
+
+         @foreach (string coffeeType in ViewBag.coffeeTypes)
+         {
+            <div>
+               <h3>C# Types</h3>
+               <ol>
+                  <li>@coffeeType</li>
+               </ol>
+            </div>
+         }
+
+   Which option produces:
+
+   #. One heading, 4 ordered lists, and 4 coffee names (each name labeled as "1")?
+   #. One heading, one ordered list, and 4 coffee names (with the names labeled
+      1, 2, 3...)?
+   #. 4 headings, 4 ordered lists, and 4 coffee names (each name labeled as "1")?
```

Comment:
After reading the whole page, this feels disjointed. The questions serve better as concept checks. I would shorten this warning to just reflect that location matters and give one example of something looking off because a dev chose the wrong locations in their template. Then move the code blocks next to the questions in the concept checks. That might also make it easier to import into Canvas.